joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/IdentifierLocalTests.scala

```diff
@@ -0,0 +1,83 @@
+package io.joern.rubysrc2cpg.passes.ast
+
+import io.joern.rubysrc2cpg.testfixtures.RubyCode2CpgFixture
+import io.shiftleft.semanticcpg.language._
+
+class IdentifierLocalTests extends RubyCode2CpgFixture {
+
+  val cpg = code("""
+      |def method1()
+      |  x = 1
+      |  x = 2
+      |end
+      |
+      |def method2(x)
+      |  x = 2
+      |end
+      |
+      |def method3(x)
+      |  y = 0
+      |
+      |  if true
+      |    innerx = 0
+      |    innery = 0
+      |
+      |    innerx = 1
+      |    innery = 1
+      |  end
+      |
+      |  x = 1
+      |  y = 1
+      |end
+      |
+      |""".stripMargin)
+
+  // TODO: Need to be fixed.
+  "be correct for local x in method1" ignore {
+    val List(method) = cpg.method.nameExact("method1").l
+    method.block.ast.isIdentifier.l.size shouldBe 2
+    val List(indentifierX, _) = method.block.ast.isIdentifier.l
+    indentifierX.name shouldBe "x"
+
+    val localX = indentifierX._localViaRefOut.get
```

Comment:
One can also go via the `Local` node via `referringIdentifiers`

Reply:
Do you mean to add a unit test for traversal from the Local node to the identifier?

Reply:
Not necessarily a unit test, but depending on how you query it, the intended API is from `LOCAL -> IDENTIFIER` using `referringIdentifiers` or from `IDENTIFIER -> LOCAL` via `refsTo` which is perhaps less explicit about the node it returns.

Reply:
Added `refsTo` check for one of the unit test and pushed the respective test. `refsTo` also not working.